### PR TITLE
[1548] Alert if retrieve trn or qts/award jobs time out

### PR DESCRIPTION
**Pull request by JonF1**
### Context
https://trello.com/c/cetXU8sR/1548-alert-if-retrievetrn-qts-jobs-time-out
Jobs were timing out silently
### Changes proposed in this pull request
* Adds module with `send_message_to_slack` method that uses `SlackNotifierService`. The method is then used in both jobs, module is included. When polling takes longer than 4 days, the job ceases to continue polling and instead uses that method to send a failure message to #twd_publish_register_tech channel. 
* Adds extra method stub and assertions for job specs
Screenshot 2021-04-28 at 19 52 41

### Guidance to review
Create new trainee in console and set required values. Set `trainee.submitted_for_trn_at` and/or `trainee.recommended_for_award_at` to 4 or more days ago, save trainee. Then run job(s)

### CODE REVIEW

**Comment by JonF1** [comment]
[[M] Alert if RetrieveTRN/QTS jobs time out](https://trello.com/c/cetXU8sR/1548-m-alert-if-retrievetrn-qts-jobs-time-out)

File: app/jobs/retrieve_award_job.rb [hunk 1]
```diff
@@ -13,6 +14,7 @@ def perform(trainee)
       trainee.award!
     elsif continue_polling?(trainee)
       self.class.set(wait: Settings.jobs.poll_delay_hours.hours).perform_later(trainee)
+    elsif send_message_to_slack(trainee, self.class)
```
Comment (mooktakim) [inline app/jobs/retrieve_award_job.rb:17]: This should be an `else`.
`elsif` is conditional. In this case we just want to run the code after all the other `if/elsif` has failed.

Also best to send `self.class.name`, which is a string.

File: app/jobs/retrieve_trn_job.rb [hunk 2]
```diff
@@ -13,6 +14,7 @@ def perform(trainee)
       trainee.trn_received!(trn)
     elsif continue_polling?(trainee)
       self.class.set(wait: Settings.jobs.poll_delay_hours.hours).perform_later(trainee)
+    elsif send_message_to_slack(trainee, self.class)
```
Comment (mooktakim) [inline app/jobs/retrieve_trn_job.rb:17]: Similar here to retrieve_award_job comment.

File: app/jobs/retrieve_award_job.rb (hunk 1 above)
Comment (JonF1) [reply app/jobs/retrieve_award_job.rb]: cheers, I've fixed

**Review by mooktakim — APPROVED** [review]
Looks good

**Pull request merged**